ledger/indexer-benchmark/src/app/scala/ledger/indexerbenchmark/Main.scala

```diff
@@ -3,92 +3,34 @@
 
 package com.daml.ledger.indexerbenchmark
 
-import java.nio.file.{Files, Paths}
-import java.util.concurrent.atomic.AtomicLong
-
-import akka.NotUsed
 import akka.actor.ActorSystem
 import akka.stream.Materializer
-import akka.stream.scaladsl.{Sink, Source}
-import com.codahale.metrics.MetricRegistry
-import com.daml.ledger.api.health.{HealthStatus, Healthy}
-import com.daml.ledger.configuration.LedgerId
+import akka.stream.scaladsl.Sink
 import com.daml.ledger.offset.Offset
-import com.daml.ledger.participant.state.kvutils.api.{
-  KeyValueParticipantStateReader,
-  LedgerReader,
-  LedgerRecord,
-}
-import com.daml.ledger.participant.state.kvutils.export.ProtobufBasedLedgerDataImporter
-import com.daml.ledger.participant.state.kvutils.{KVOffsetBuilder, Raw}
 import com.daml.ledger.participant.state.v2.Update
 import com.daml.logging.LoggingContext.newLoggingContext
-import com.daml.metrics.Metrics
+import com.daml.platform.indexer.ha.EndlessReadService
 
 import scala.concurrent.{ExecutionContext, Future}
 
 object Main {
-  def main(args: Array[String]): Unit =
-    IndexerBenchmark.runAndExit(args, name => loadLedgerExport(name))
-
-  private[this] def loadLedgerExport(config: Config): Future[Iterator[(Offset, Update)]] = {
-    val path = Paths.get(config.updateSource)
-    if (!Files.exists(path)) {
-      throw new RuntimeException(s"Input file $path does not exist")
-    }
-    val importer = ProtobufBasedLedgerDataImporter(path)
-
-    val offsetBuilder = new KVOffsetBuilder(0)
-    val dataSource: Source[LedgerRecord, NotUsed] = Source
-      .fromIterator(() => importer.read().iterator)
-      .statefulMapConcat { () =>
-        val nextOffset = new AtomicLong(0)
-
-        { case (_, writeSet) =>
-          writeSet.map { case (key, value) =>
-            val offset = offsetBuilder.of(nextOffset.getAndIncrement())
-            val logEntryId = Raw.LogEntryId(key.bytes) // `key` is of an unknown type.
-            LedgerRecord(offset, logEntryId, value)
-          }
-        }
-      }
-
-    val keyValueSource = new LedgerReader {
-      override def events(offset: Option[Offset]): Source[LedgerRecord, NotUsed] =
-        if (offset.isDefined) {
-          Source.failed(
-            new IllegalArgumentException(
-              s"A read offset of $offset is not supported. Must be $None."
-            )
-          )
-        } else {
-          dataSource
-        }
-
-      override def currentHealth(): HealthStatus = Healthy
+  val DefaultUpdateCount = 500000L
```

Comment:
Maybe it's worth changing the type of `Config.updateCount` from `Option[Long]` to `Long` and set this value as a default.

Reply:
I think for the LR jobs which would be executed from oem it would break previous implementation as there we do not want to limit how many updates to load and instead load ALL events from the file.

Reply:
Please correct me if I'm wrong, but it seems that we do `.take(config.updateCount.getOrElse(DefaultUpdateCount))` which will trim the stream anyway, so it's just about moving things around.

Reply:
In oem repo there will be a completely different Main using the same Config object and IndexerBenchmark class which would not trim the stream.

Reply:
Ok, I got confused by looking at the draft in the oem repo - there's trimming in place but if it's not going to be used then it makes sense to keep the parameter optional. 👍

Reply:
It is there in place, but it might not be used, i.e. default would be `Long.MaxValue` and it's not a proper default for the use case in this repo as stream is endless.